### PR TITLE
Adding code for hardening the local RKE cluster in a HA setup

### PR DESCRIPTION
**Pull request by sowmyav27**

### CODE REVIEW

File: tests/validation/tests/v3_api/common.py [hunk 1]
```diff
@@ -2962,3 +2917,63 @@ def update_and_validate_kdm(kdm_url, admin_token=ADMIN_TOKEN,
...
+        network_policy_file = DATA_SUBDIR + "/default-allow-all.yaml"
+        account_update_file = DATA_SUBDIR + "/account_update.yaml"
+        items = execute_kubectl_cmd("get namespaces -A", 
+        kubeconfig=kubeconfig_path)["items"]
+        all_ns = [item["metadata"]["name"] for item in items]
+        for ns in all_ns:
+            execute_kubectl_cmd("apply -f {0} -n {1}".
+                                format(network_policy_file, ns), 
+                                kubeconfig=kubeconfig_path)
+        namespace = ["default", "kube-system"]
```
Comment (izaac) [inline tests/validation/tests/v3_api/common.py:2974]: I went through the hardening guide once more and I think the scripts there apply the default service account patch to all namespaces not only `default` and `kube-system`.

**Review by izaac — CHANGES_REQUESTED** [review]

**Comment by izaac** [comment]
When i tried to setup this environment manually I observed that after installing Rancher fleet and other components create their namespaces after Rancher is running. Are we going to wait for these other resources to get created to re-apply the service account and network policies patches?

File: tests/validation/tests/v3_api/common.py (hunk 1 above)
Comment (sowmyav27) [reply tests/validation/tests/v3_api/common.py]: changed code to include all ns. Verified the hardened scan still `passes`

**Review by izaac — DISMISSED** [review]
LGTM

File: tests/validation/tests/v3_api/common.py [hunk 2]
```diff
@@ -2962,3 +2917,61 @@ def update_and_validate_kdm(kdm_url, admin_token=ADMIN_TOKEN,
...
+                docker_run_cmd = \
+                get_custom_host_registration_cmd(client,
+                                                 cluster,
+                                                 node_roles[i],
+                                                 aws_node)
+                aws_node.execute_command(docker_run_cmd)
+            i += 1
+    elif profile == 'rke-cis-1.5':
+        for aws_node in aws_nodes:
+            aws_node.execute_command("sudo sysctl -w vm.overcommit_memory=1")
```
Comment (bmdepesa) [inline tests/validation/tests/v3_api/common.py:2942]: Is it possible to move these hardening commands to a script and either run the script as a whole or line by line?

**Comment by sowmyav27** [comment]
Closing this for https://github.com/rancher/rancher/pull/35023

**Pull request closed without merging**